File: src/2920_scale/scale.py
# https: // www.acmicpc.net/problem/2920

a = list(map(int, input().split(' ')))

asc = True
desc = True

for i in range(1, 8):
    if a[i] > a[i - 1]:
        desc = False
    elif a[i] < a[i - 1]:
        asc = False

if asc:
    print('asc')
elif desc:
    print('desc')
else:
    print('mix')

File: src/2920_scale/scale.js
//https://www.acmicpc.net/problem/2920

const ASC = "ascending";
const DESC = "descending";
const MIXED = "mixed";

function music(input) {
  const values = input.split(" ");
  const valInt = values.map((value) => parseInt(value));

  let asc = true;
  let desc = true;

  for (let i = 1; i < valInt.length; i++) {
    if (valInt[i] > valInt[i - 1]) {
      desc = false;
    } else if (valInt[i] < valInt[i - 1]) {
      asc = false;
    }
  }

  if (asc) {
    console.log(ASC);
  } else if (desc) {
    console.log(DESC);
  } else {
    console.log(MIXED);
  }
}

music("1 2 3 4 5 6 7 8");
music("8 7 6 5 4 3 2 1");
music("1 3 2 4 5 6 7 8");
music("8 3 2 4 5 6 7 1");
music("1 2 3 4 5 6 8 7");
music("8 7 6 5 4 3 1 2");
